content/2022/robot-valjak.md

```diff
@@ -2,3 +2,208 @@
+
+1. Koji svojim naponom od 0V ili 5V odredjuje smer i 
+2. Koji koristeci PWM (Pulse Width Modulation) brzim paljenjem i gašenjem kontroliše napon koji je doveden do motora u opsegu od 0-12V
+
+JGA25370 je DC motor koji radi na naponu od 12V i maksimalnoj struji od 2A. Motor u sebi poseduje reduktor koji umanjuje broj obrtaja motora u sekundi ali povecava obrtni momenat gde je broj obrtaja $RPM_{max} = 250$ obrtaja u minutu i obrtni momenat $T = 1.4kg/cm$. Ovaj motor je jedina komponenta koja pokreće robota.
+
+##### E38S6G5 Opticki inkrementalni enkoder
+
+E38S6G5 je optički enkoder koji se koristi za merenje pređenog puta robota. E38S6G5 ima rezoluciju od 600 pulseva po obrtaju osovine što nam daje preciznost od  0.6 stepeni ili ~0.01rad. Princip rada optičkog enkodera je da se analizom izlaznih talasa može odrediti da li se enkoder okrenuo u pozitivnom ili negativnom smeru. U konkretnom slučaju na pin 3 mikrokontrolera je povezan prvi od dva izlaza enkodera, gde se prilikom svake rastuće ivice ovog pina mikrokontroler beleži napon na drugom izlazu enkodera. U slučaju da je napon pozitivan mikrokontroler to belezi kao obrt u pozitivnom smeru, a u slučaju da je napon negativam mikrokontroler to belezi kao obrt u negativnom smeru. Kada za svaki ovaj dogadjaj od nekog brojaša saberemo ili oduzmemo 1 mozemo dobiti ukupni ugao za koji se enkoder zarotirao.
+
+## Istraživanje i rezultati
+
+Tokom rada na projektu, uspešno je realizovana hardverska implementacija robota i prikupljeni su podaci o tome kako sistem reaguje na uzastopne odskočne odzive. Tokom istraživanja su izmerena tri odskočna odziva koja redom odgovaraju transfer funkcijama ugla inklinacije klatna u odnosu na podlogu u zavisnosti od napona motora, ugaone brzine klatna u odnosu na podlogu u zavisnosti od ulaznog napona motora i ugaonog pređenog puta robota u zavisnosti od ulaznog napona motora.
+
+Prilikom dizajniranja klatna valjka centar mase korpe i komponenti (baterije, arduino i ostali moduli) se ne nalazi u osnosimetričnoj ravni korpe već je blago izbačen sa strane što dovodi do toga da nula u merenju inklinacije ugla nije prava nula. Takođe, ovome je potpomogla i činjenica da se na DC motoru nalazi reduktor zbog kog ne postoji prava nula već postoji interval uglova koji mogu biti prava nula. Za male uglove moment zemljine teže na klatno se izjednačava sa momentom sile koju reduktor proizvodi na klatno. Drugim rečima, javalja se mali interval oko prave nule za koje se klatno nalazi u indiferentnoj ravnoteži i u kojoj krajnji položaj zavisi isključivo od početnog impulsa. Ukoliko je veći početni impuls veća je i greška merenja ugla inklinacije. Ovaj fenomen se može predstaviti na grafiku ispod gde se vidi da robot konstantno *driftuje* u jednu stranu. Za dalji rad na ovom robotu treba da se poravna centar mase sistema kao i da se poveća masa klatna kako bi se smanjio opseg uglova koji mogu biti prava nula i na taj način postići preciznija merenja.
```

Comment:
1. "nula u merenju inklinacije ugla nije prava nula" -> "merenje ugla ne bude centrirano oko nule" ("prava nula" zvuči čudno)
2. "Takođe, ovome je potpomogla i činjenica da se na DC motoru nalazi reduktor zbog kog ne postoji prava nula već postoji interval uglova koji mogu biti prava nula" Ovo ima još manje smisla. Takođe, reduktor ne utiče na merenje, on utiče na fiziku koju niste modelovali, a na merenje utiče samo diskretizacija ugla u enkoderu.
3. "Za male uglove moment zemljine teže na klatno se izjednačava sa momentom sile koju reduktor proizvodi na klatno" zašto samo za male uglove? Ovo može da bude pod bilo kojim uglom. Takođe gde je nestao moment usled statičkog trenja?
4. "u kojoj krajnji položaj zavisi isključivo od početnog impulsa" Ovo mi nije najjasnije
5. "Ukoliko je veći početni impuls veća je i greška merenja ugla inklinacije" Kako dinamika robota utiče na merenje uglova? Ovo ne bi trebalo da je tačno
6. "Ovaj fenomen se može predstaviti na grafiku ispod gde se vidi da robot konstantno *driftuje* u jednu stranu" Da li je ovo sigurno tačno? Da nije ovo nešto slično random walk-u, tj. da li ako se eksperiment ponovi dovoljan broj puta, na različitim podlogama pod različitim uglovima vidimo ovaj efekat? Takođe, merenje ugla i ovo što zovete "prava nula" ne utiče na merenje koje ste pokazali zato što nigde ne koristite to merenje.
7. Objasnite ovaj grafik malo bolje, šta je pobuda i gde. Kako je pređeni put robota u radianima?
8. "Za dalji rad na ovom robotu treba da se poravna centar mase sistema kao i da se poveća masa klatna kako bi se smanjio opseg uglova koji mogu biti prava nula i na taj način postići preciznija merenja." - Ne verujem da je ovo zaključak iz onoga što ste napisali i ovog komentara. Ako reduktor unosi nelinearnu komponentu u sistem, to ne može da se izbegne, što znači da treba ili da se zanemari ili doda u algoritam za kontrolu.

Reply:
2) Utiče na merenje zato što kada robot miruje klatno nije vertikalno i onda iz očitavanja može da se zaključi da se robot kreće
3) stvar je o tome da iako nema napona na motoru klatno nece mirovati vertikalno već pod nekim uglom. Taj neki ugao ne mora matematički da bude mali, vec da samo postoji ali da nije bas 90stepeni
4) ako je klatno u početku vertikalno i ako ga slabo gurneš on će se pomeriti za neki ugao. Ako ga gurneš malo jače on će se zaustaviti na nekom višem uglu sve dok ga ne gureneš dovoljno jako da ne izađe i te zone.

Reply:
5. Ovaj deo nema veze sa robotom dok se kreće već kažem da klatno kada bismo gurnuli ono se ne bi vratilo u prvobitni položaj već bi ostalo pod nekim uglom. Ja sam reko da je taj ugao veći što je impuls veći sve dok impuls ne postane dovoljno velik da zapravo obrne klatno
6. Taj random walk je uvek bio na istoj strani robota i svaki put se pojavljivao. Što se tiče prave nule ona ne utiče ninašta ali nam je predstavljala problem u implementaciji PID-ova zato i stoji tu to je kao naš zaključak šta bi tebalo u buduće da se radi
8. Reduktor unosi nelinearnu komponentu ali je ideja da se klatno oteža i izbalansira kako bi se smanjio efekat reduktora